highlevel/robot/controller/strategy.py

```diff
@@ -47,21 +50,71 @@
     (Vector2(300, 1450), True),
 ]
 
+PATH_MIRRORED = [(Vector2(pos.x, 2000 - pos.y), reverse)
+                 for pos, reverse in PATH]
+
+
+class ControlMode(Enum):
+    """
+    Used to describe control modes of the motor board.
+    POSITION: Motor board tries to move the wheels at each wheel's target position.
+    SPEED: Motor board tries to move the wheels at each wheel's target speed.
+    """
+    POSITION = 'POSITION'
+    SPEED = 'SPEED'
+
 
 class StrategyController:
     """
     The strategy controller holds the high level algorithm executed by the robot.
     """
-    def __init__(self, motion_controller: MotionController):
-        self.motion_controller = motion_controller
+    def __init__(self, trajectory_controller: TrajectoryController,
+                 configuration: Configuration):
+        self.configuration = configuration
+        self.trajectory_controller = trajectory_controller
+
+    async def set_mode(self, mode: ControlMode) -> None:
+        """
+        Sets the control mode of the motor board.
+        """
+        if mode == ControlMode.SPEED:
+            await self.trajectory_controller.motion_controller.motor_gateway.set_pid_speed(
+                self.configuration.pid_constants_speed_left.k_p,
+                self.configuration.pid_constants_speed_left.k_i,
+                self.configuration.pid_constants_speed_left.k_d,
+                self.configuration.pid_constants_speed_right.k_p,
+                self.configuration.pid_constants_speed_right.k_i,
+                self.configuration.pid_constants_speed_right.k_d,
+            )
+            await self.trajectory_controller.motion_controller.motor_gateway.set_control_mode(
+                True, False)
+        elif mode == ControlMode.POSITION:
+            await self.trajectory_controller.motion_controller.motor_gateway.set_pid_position(
+                self.configuration.pid_constants_position_left.k_p,
+                self.configuration.pid_constants_position_left.k_i,
+                self.configuration.pid_constants_position_left.k_d,
+                self.configuration.pid_constants_position_right.k_p,
+                self.configuration.pid_constants_position_right.k_i,
+                self.configuration.pid_constants_position_right.k_d,
+            )
+            await self.trajectory_controller.motion_controller.motor_gateway.set_control_mode(
+                False, True)
 
     async def run(self) -> None:
         """
         Run the strategy.
         """
-        for vec, reverse in PATH:
```

Comment:
So, if we are removing this, why are we keeping PATH and PATH_MIRRORED?

I'd prefer to keep this

Reply:
You can revert to it if needed.
There is no strategy in the robot anyway, should we remove the strategy controller too?

Reply:
There is a stub strategy (this long list of positions).
It's just very rough

I like the fact that I can use this long running scenario for testing